### PR TITLE
Created API's for Get Open and Closed Issues

### PR DESCRIPTION
**Pull request by Uttkarsh-raj**
This pr is regarding getting the open and closed issues for a domain. The domain must be sent as a query in the API call.
## Request:
```
http://localhost:8000/open-issues/?domain=http://www.blt.com
``` 
## Response:
```
{
  "count": 1,
  "result": [
    {
      "id": 3,
      "user": {
        "id": 1,
        "username": "admin"
      },
      "url": "https://youtu.be/kopl",
      "description": "sdfasfdsaf",
      "markdown_description": "sdfasdfsfsdfsa",
      "label": 0,
      "views": 1,
      "verified": false,
      "score": null,
      "status": "open",
      "user_agent": null,
      "ocr": "",
      "screenshot": "/media/screenshots/Screenshot_44.png",
      "closed_date": null,
      "github_url": null,
      "created": "2024-05-10T19:12:27.348276Z",
      "modified": "2024-05-11T13:04:08.699875Z",
      "is_hidden": false,
      "rewarded": 0,
      "reporter_ip_address": null,
      "cve_id": null,
      "cve_score": null,
      "hunt": null,
      "domain": 2,
      "closed_by": null,
      "team_members": [
        1
      ]
    },
  ]
}
```
Screenshot (104)
Screenshot (105)

Similar for the Closed API.

Screenshot (106)

### CODE REVIEW

File: website/api/views.py
Comment (arkid15r) [inline website/api/views.py:1]: Great work figuring out the `viewsets`! 

It looks like your Open/Closed issue views separation approach isn't going along with DRY (do not repeat yourself)  principle. I suggest using a single viewset and branch logic purely based on the URL parameter specifying issue's status (open/closed). Something like

`path("issues/<str:status>", IssuesViewSet.as_view({"get": "get"}), name="issues")`

Here you need to make sure the URL doesn't clash with existing mapping and it'd also be great to narrow down the status options to ["open", "closed"].

File: website/api/views.py [hunk 1]
```diff
@@ -617,3 +619,39 @@ class CompanyViewSet(viewsets.ModelViewSet):
...
+
+
+class OpenIssuesViewSet(viewsets.ModelViewSet):
+    queryset = Issue.objects.all()
+    serializer_class = IssueSerializer
+
+    def get(self, request, *args, **kwargs):
+        try:
+            domainUrl = request.GET.get("domain")
+            domainData = Domain.objects.get(url=domainUrl)
```
Comment (arkid15r) [inline website/api/views.py:631]: Snake case please

File: website/api/views.py [hunk 2]
```diff
@@ -617,3 +619,39 @@ class CompanyViewSet(viewsets.ModelViewSet):
...
+            data = Issue.objects.filter(domain=domainData.id, status="open")
+            serializer = IssueSerializer(data, many=True)
+            return Response({"count": len(serializer.data), "result": serializer.data}, status=200)
+        except ObjectDoesNotExist:
```
Comment (arkid15r) [inline website/api/views.py:635]: You could narrow it down to `Issue.DoesNotExist` exception

**Review by arkid15r — COMMENTED** [review]
Nicely done, please find my comments below:

**Comment by Uttkarsh-raj** [comment]
> You can accomplish this by adding a filter to the existing endpoint

something like this issues/?status="open"&domain="www.blt.com" ?

**Comment by Uttkarsh-raj** [comment]
Please review this approach i tried. The output is the required one but i want to know if this approach is correct or not.
Screenshot (107)
Screenshot (108)
Screenshot (109)
Screenshot (110)

**Comment by Uttkarsh-raj** [comment]
> This looks right

Plz review the code too...Thanks..

**Comment by Uttkarsh-raj** [comment]
> You can update the existing code here https://github.com/OWASP-BLT/BLT/blob/main/website/api/views.py

Can you specify the changes needed.

**Comment by arkid15r** [comment]
> Can you specify the changes needed.

Try playing with  https://www.django-rest-framework.org/api-guide/filtering/

**Comment by Uttkarsh-raj** [comment]
> Try playing with https://www.django-rest-framework.org/api-guide/filtering/
Please review these changes. Thank you.

**Review by DonnieBLT — CHANGES_REQUESTED** [review]

**Comment by Uttkarsh-raj** [comment]
@DonnieBLT made these changes. Thanks for this. Any other changes needed?

**Review by arkid15r — COMMENTED** [review]
It looks much better now, @Uttkarsh-raj!

Thanks for addressing all comments!

LGTM

**Review by DonnieBLT — APPROVED** [review]

**Comment by Uttkarsh-raj** [comment]
> It looks much better now, @Uttkarsh-raj!
> Thanks for addressing all comments!
> LGTM

Thanks for all the reviews have really learnt a lot from this...

**Pull request merged**